src/main/java/org/jabref/gui/maintable/BibEntryTableViewModel.java

```diff
@@ -108,15 +108,12 @@ public BibEntry getEntry() {
     }
 
     public ObservableValue<String> getFields(OrFields fields) {
-        ObservableValue<String> value = fieldValues.get(fields);
-        if (value != null) {
-            return value;
-        }
-
         ArrayList<Observable> observables = new ArrayList<>(List.of(entry.getObservables()));
+        Optional<BibEntry> referenced = fieldValueFormatter.getValue().getReferencedEntry(entry);
+        referenced.ifPresent(bibEntry -> observables.addAll(List.of(bibEntry.getObservables())));
```

Comment:
I don't know enough of JavaFX to tell you for sure how to address this. I'd suggest not removing the caching (`fieldValues.get(fields);`) and instead deal with the dependencies differently.

* You can extend `StringBinding` and use [.bind](https://openjfx.io/javadoc/16/javafx.base/javafx/beans/binding/StringBinding.html#bind(javafx.beans.Observable...)) and [.unbind](https://openjfx.io/javadoc/16/javafx.base/javafx/beans/binding/StringBinding.html#unbind(javafx.beans.Observable...)) to "manually" deal with the dependencies. (this is probably not a good approach)
* You can extend a different binding (`ObjectBinding`?) and have it handle changes to the referenced entry. In this scenario, the `StringBinding` has this `ObjectBinding` among its dependencies instead of the referenced bibentry's `getObservables()`.
* At least superficially this seems like a good application for [EasyBind](https://github.com/tobiasdiez/EasyBind) because, 
	1. You can create an optional binding, e.g., `entry.getFieldBinding(StandardField.CROSSREF)`
		1. This means you can create an `OptionalBinding<BibEntry> referencedBibEntry`
	2. It should be possible to keep an [observableArrayList with extractor](https://openjfx.io/javadoc/16/javafx.base/javafx/collections/FXCollections.html#observableArrayList(javafx.util.Callback)) (`FXCollections.observableArrayList(BibEntry::getObservables)`) updated with the `ifValuePresent` method of the `OptionalBinding<BibEntry>`. That should keep the list up-to-date when the referenced `BibEntry` is changed/removed and when the observables in `bibEntry.getObservables` are modified
	3. This list should now be usable as a dependency

Note that the two last approaches should be used or called from the constructor since they will update on changes to the referenced `BibEntry`.

I'd appreciate it if someone with more experience of JavaFX chimes in with opinions 😛

Reply:
Discussion:

```bibtex
@artcile{a,
crossref={b}
}

@book{b}
```

to

```bibtex
@artcile{a,
crossref={c}
}

@book{b}

@book{c}
```

Changes to c are not reflected accordingly.